src/libpriv/rpmostree-core.c

```diff
@@ -2696,38 +2828,54 @@ rpmostree_context_assemble_tmprootfs (RpmOstreeContext      *self,
         return FALSE;
     }
 
+  /* We're completely disregarding how rpm normally does upgrades/downgrade here. rpm
+   * usually calculates the fate of each file and then installs the new files first, then
+   * removes the obsoleted files. So the TR_ADDED element for the new pkg comes *before* the
+   * TR_REMOVED element for the old pkg. This makes sense when operating on the live system.
+   * Though that's not the case here, so let's just take the easy way out and first nuke
+   * TR_REMOVED pkgs, then checkout the TR_ADDED ones. This will probably start to matter
+   * though when we start handling e.g. file triggers. */
```

Comment:
Offhand I think there's going to be little harm in just running all of the triggers at the end.